src/main/java/edu/harvard/iq/dataverse/TemplatePage.java

```diff
@@ -183,6 +184,10 @@ public String save(String redirectPage) {
         Long createdId = new Long(0);
         Template created;
         try {
+            Iterator<DatasetField> dsfItSort = template.getDatasetFields().iterator();
```

Comment:
This looks like a fix, but - wouldn't the other checking in tidyUpFields also make sense before saving the template? If so, perhaps there's also an opportunity to refactor - make tidyUpFields() take dsv.getDatasetFields / template.getDatasetFields as the parameter (perhaps moving that method to a util class).

Also - seeing that the current tidyUpFields is called from other Commands - should the tidying here also be done in the template commands (on principle/to help if there's a future template API where order also needs to be defined)?

Reply:
Hi @qqmyers,

Yes, my first idea was to run the tidyUpFields() method, however, tidyUpFields also removes all empty fields and this creates another error when editing a template,

![image](https://user-images.githubusercontent.com/31973211/140989291-4cf76558-6411-4804-979c-4f01be8e6265.png)

If you save an author without affiliation or identifier, when you try to edit it, these fields are no longer displayed. What do you think?

Reply:
Hmm, but that doesn't happen in dataset metadata. Probably some other method getting called to compensate in datasets. While it would be nice to have templates and datasets work the same, I don't know that its worthwhile to ask you to go investigating when what you've put in the PR is a fix for the issue. Unless somebody else has info or has a reason that not adding the empty field removal and trailing space removal that happens in tidyUpFields() is going to a big deal, I'd say it's OK as is - perhaps just adding a note in the code about this issue/difference with datasets.

Reply:
So the method that makes sure tou don't have missing fields when editing a dataset is DatasetVersionUI (it also looks like it might handle the orderingsinc it has a sort:
    private List<DatasetField> initDatasetFields(boolean createBlanks) {

it is private because it's called by:
    public DatasetVersionUI  initDatasetVersionUI(DatasetVersion datasetVersion, boolean createBlanks) {

which does more than the template would need. 

But to avoid technical debt, we'd probably want to rely on trying to use these (possibly needing to be centralized) methods and tidyUpFields.